app/src/main/java/com/github/se/project/ui/lesson/CompletedLesson.kt

```diff
@@ -1,3 +1,219 @@
-package com.github.se.project.ui.lesson
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Star
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag // Import for test tags
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import com.github.se.project.model.lesson.Lesson
+import com.github.se.project.model.lesson.LessonViewModel
+import com.github.se.project.model.profile.ListProfilesViewModel
+import com.github.se.project.model.profile.Profile
+import com.github.se.project.model.profile.Role
+import com.github.se.project.ui.components.DisplayLessonDetails
+import com.github.se.project.ui.navigation.NavigationActions
+import com.github.se.project.utils.formatDate
 
-class CompletedLessonScreen {}
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CompletedLessonScreen(
+    listProfilesViewModel: ListProfilesViewModel,
+    lessonViewModel: LessonViewModel,
+    navigationActions: NavigationActions
+) {
+  val lesson by lessonViewModel.selectedLesson.collectAsState()
+  val currentUserProfile by listProfilesViewModel.currentProfile.collectAsState()
+
+  if (lesson == null || currentUserProfile == null) {
+    Text(
```

Comment:
Consider adding some comments for better understanding by others.

Reply:
alright I'll be more careful next time, thanks for the feedback!